app/controllers/companies_controller.rb

```diff
@@ -1,5 +1,26 @@
 # frozen_string_literal: true
 
 class CompaniesController < ApplicationController
-  before_action :authenticate_company!, only: %i[index interested_people]
+  before_action :set_company, only: %i[show edit update destroy]
+  before_action :authenticate_company!, only: %i[index interested_people show]
+
+  def index
+    @companies = Company.all
+  end
+
+  def show
+    @company_reviews = CompanyReview.where(company_id: @company.id).order('created_at DESC')
+
+    @avg_review = if @company_reviews.blank?
```

Comment:
@company_reviews is giving you blank array?

Reply:
yes in that case when there are no company reviews for the given company